auth/TESLA/Receiver.m

```diff
@@ -0,0 +1,135 @@
+classdef Receiver < handle
+    % RECIEVER Summary of this class goes here
+    %   Detailed explanation goes here
+
+    properties
+        key_chain_function = @HashingWrappers.sha_256
+        mac_signing_function = @HashingWrappers.hmac_sha_256
+        message_table
+        reciever_key_chain = cell(1, 100)
+        next_key_chain = cell(1, 100)
+        verified_block = cell (1, 5)
+        starting_hash
+        mt50_table
+        missing_mt50 = {}
+    end
+
+    methods
+
+        function obj = Receiver(hp, hp2)
+            obj.reciever_key_chain{1} = hp;
+            obj.next_key_chain{1} = hp2;
+            obj.message_table = containers.Map('KeyType', 'double', 'ValueType', 'any');
+            obj.mt50_table = containers.Map('KeyType', 'double', 'ValueType', 'any');
+        end
+
+        function add_message(obj, message, time)
+            to_add = cell(1, 2);
+            to_add{1} = message;
```

Comment:
Make this either a struct or a nested class. Either will enforce that the `message` and `authenticated?` have a specific type. Cell arrays have not requirement on type...which makes this choice brittle.